docs/pages/setup/guides/terraform-provider.mdx

```diff
@@ -0,0 +1,236 @@
+  tar -zxvf terraform-provider-teleport-v(=teleport.version=)-linux-amd64-bin.tar.gz -C ${HOME?}/.terraform.d/plugins/gravitational.com/teleport/teleport/(=teleport.version=)/linux_amd64
+  ```
+  </TabItem>
+
+  <TabItem label="MacOS">
+  ```bash
+  mkdir -p ${HOME?}/.terraform.d/plugins/gravitational.com/teleport/teleport/(=teleport.version=)/darwin_amd64
+  ```
+  </TabItem>
+</Tabs>
+  
+# Step 2/4. Create a terraform user
+
+Put the following content into `terraform.yaml`:
+
+```
+kind: role
+metadata:
+  name: terraform
+spec:
+  allow:
+    rules:
+      - resources: ['user', 'role', 'token', 'trusted_cluster', 'github', 'oidc', 'saml']
+        verbs: ['list','create','read','update','delete']
+version: v3
+---
+kind: user
+metadata:
+  name: terraform
+spec:
+  roles: ['terraform']
+version: v2
+```
+
+Run:
+
+```bash
+tctl create terraform.yaml
+tctl auth sign --format=file --user=terraform --out=terraform-identity --ttl=10h
+```
+
+<Admonition type="note">
+Clients missing impersonation privileges when trying to use `tctl auth sign`,
+will get the following error:
+
+```bash
+ERROR: access denied: impersonation is not allowed
+```
+
+Create the following file with role: `terraform-impersonator.yaml`:
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: terraform-impersonator
+spec:
+  # SSH options used for user sessions 
+  options:
+    # max_session_ttl defines the TTL (time to live) of SSH certificates 
+    # issued to the users with this role.
+    max_session_ttl: 10h
+
+  # allow section declares a list of resource/verb combinations that are
+  # allowed for the users of this role. by default nothing is allowed.
+  allow:
+    impersonate:
+      users: ['terraform']
+      roles: ['terraform']
+
+  # the deny section uses the identical format as the 'allow' section.
+  # the deny rules always override allow rules.
+  deny:
+    node_labels:
+      '*': '*'
+```
+
+```bash
+tctl create terraform-impersonator.yaml
+```
+
+Assign this role to the current user. Re-login to assume the new role and try
+to issue certificate for terraform user again.
+
+</Admonition>
+
+## Step 3/4. Create Terraform configuration
+
+Create a `main.tf` terraform file:
+
+```
+terraform {
+  required_providers {
+    teleport = {
+      version = ">= (=teleport.version=)"
+      source  = "gravitational.com/teleport/teleport"
+    }
+  }
+}
+
+provider "teleport" {
+  # Update addr to point to Teleport Auth/Proxy
+  addr               = "proxy.example.com:443"
+  identity_file_path = "terraform-identity"
+}
+
+resource "teleport_role" "terraform-test" {
+  metadata {
+    name        = "terraform-test"
+    description = "Terraform test role"
+    expires     = "2022-10-12T07:20:50.52Z"
+    labels = {
+      example  = "yes"
+    }
+  }
+  
+  spec {
+    options {
+      forward_agent           = false
+      max_session_ttl         = "7m"
```

Comment:
I'm assuming the `7m` value is intentional but thought I'd point that out since that seems like a pretty short interval.

Reply:
good point, fixed